screens/CheckoutScreen.js

```diff
@@ -33,98 +32,105 @@ export default class CheckoutScreen extends React.Component {
-      cart: update(prevState.cart, { [product.id]: { quantity: { $set: quantity } } }),
-      totalPrice: prevState.totalPrice + priceDifference
-    }));
+    const newBalance = this.state.totalBalance + priceDifference;
+    // Undoing Rewards
+    /* IF: 
+      1) Quantity is increasing
+      2) No rewards have been applied, or
+      3) The new balance is non-negative
+      No special handling */
+    if (priceDifference >= 0 || this.state.rewardsApplied === 0 || newBalance >= 0) {
+      this.setState(prevState => ({
+        cart: update(prevState.cart, { [product.id]: { quantity: { $set: quantity } } }),
+        totalBalance: prevState.totalBalance + priceDifference
+      }));
+      // Special case: negative balance when rewards have been applied, but need to be restored
+      // i.e rewardsValue * rewardsApplied > cartTotal after quantity drops
+    } else if (newBalance < 0) {
+      // MUST take absolute value first, otherwise will be incorrect
+      const rewardsToUndo = Math.ceil(Math.abs(newBalance) / rewardDollarValue);
+      // This keeps the "extra" reward UNAPPLIED by default. To swap it, take the remainder instead
+      // updatedBalance: rewardDollarValue + (- remainder) = "unapplying" an extra reward
+      this.setState(prevState => ({
+        cart: update(prevState.cart, { [product.id]: { quantity: { $set: quantity } } }),
+        totalBalance:
+          // If the remainder is less than 0, updatedBalance = rewardDollarValue + (negative) remainder
+          (prevState.totalBalance + priceDifference) % rewardDollarValue < 0
+            ? rewardDollarValue + ((prevState.totalBalance + priceDifference) % rewardDollarValue)
+            : (prevState.totalBalance + priceDifference) % rewardDollarValue,
+        rewardsApplied: prevState.rewardsApplied - rewardsToUndo
+      }));
+    }
   };
 
-  // Sets total points earned from transaction in state.
-  setTotalPoints = () => {
-    let points = 0;
-    // Iterate over lineItems in cart
-    Object.values(this.state.cart).forEach(lineItem => {
-      points += lineItem.points * lineItem.quantity;
-    });
-    points -= this.state.rewardsApplied * 500;
-    if (points < 0) {
-      console.error('Total points less than 0!');
+  // Calculates total points earned from transaction
+  // Accounts for lineItem individual point values and not allowing points to be earned with rewards daollrs
+  getPointsEarned = () => {
+    let pointsEarned = Object.values(this.state.cart).reduce(
+      (points, lineItem) => points + lineItem.points * lineItem.quantity,
+      0
+    );
+    // Customer cannot earn points with rewards dollars; assumes a reward's point multiplier per dollar is 100 pts
+    pointsEarned -= this.state.rewardsApplied * rewardDollarValue * 100;
+
+    // TODO this might be a design edge case now that rewards applied can bring the technical balance to a negative
+    if (pointsEarned < 0) {
+      if (this.state.totalBalance > 0) {
+        console.log(
```

Comment:
Would this be more helpful as an error message for us to debug :0 we would remove this in the MVP right

Reply:
Good pt!! I left it as a log because of the low likelihood but possible legal scenario in which items' multipliers are low ;o 

But ya I think we should figure out a totally diff logging system for prod anyway! (I don't know where console.logs go LOL?) will leave it for now 😅